diff --git a/node_modules/@distube/ytdl-core/lib/sig.js b/node_modules/@distube/ytdl-core/lib/sig.js
index b9d1775..d423627 100644
--- a/node_modules/@distube/ytdl-core/lib/sig.js
+++ b/node_modules/@distube/ytdl-core/lib/sig.js
@@ -59,10 +59,12 @@ const HELPER_REGEXP = `var (${VARIABLE_PART})=\\{((?:(?:${
   VARIABLE_PART_DEFINE}${SPLICE_PART}|${
   VARIABLE_PART_DEFINE}${SWAP_PART}),?\\n?)+)\\};`;
 
-const N_TRANSFORM_REGEXP = 'function\\(\\s*(\\w+)\\s*\\)\\s*\\{var' +
-  '\\s*(\\w+)=\\1\\.split\\(""\\),\\s*(\\w+)=(\\[.*?\\]);\\s*\\3\\[\\d+\\]' +
-  '(.*?try)(\\{.*?\\})catch\\(\\s*(\\w+)\\s*\\)\\s*\\' +
-  '{\\s*return"enhanced_except_([A-z0-9-]+)"\\s*\\+\\s*\\1\\s*}\\s*return\\s*\\2\\.join\\(""\\)\\};';
+  const N_TRANSFORM_REGEXP = 'function\\(\\s*(\\w+)\\s*\\)\\s*\\{' +
+  'var\\s*(\\w+)=(?:\\1\\.split\\(\"\"\\)|String\\.prototype\\.split\\.call\\(\\1,\"\"\\)),' +
+  '\\s*(\\w+)=(\\[.*?]);\\s*\\3\\[\\d+]' +
+  '(.*?try)(\\{.*?})catch\\(\\s*(\\w+)\\s*\\)\\s*\\' +
+  '{\\s*return\"enhanced_except_([A-z0-9-]+)\"\\s*\\+\\s*\\1\\s*}' +
+  '\\s*return\\s*(\\2\\.join\\(\"\"\\)|Array\\.prototype\\.join\\.call\\(\\2,\"\"\\))};';
 
 const DECIPHER_ARGUMENT = 'sig';
 const N_ARGUMENT = 'ncode';
