diff --git a/node_modules/distube/dist/index.js b/node_modules/distube/dist/index.js
index 8e827f5..9b5f275 100644
--- a/node_modules/distube/dist/index.js
+++ b/node_modules/distube/dist/index.js
@@ -769,7 +769,29 @@ var DisTubeVoice = class extends import_tiny_typed_emitter.TypedEmitter {
       }
     }).on(import_voice.VoiceConnectionStatus.Destroyed, () => {
       this.leave();
-    }).on("error", () => void 0);
+    }).on("stateChange", (oldState, newState) => {
+    if (
+      oldState.status === import_voice.VoiceConnectionStatus.Ready &&
+      newState.status === import_voice.VoiceConnectionStatus.Connecting
+    ) {
+      this.connection.configureNetworking();
+    }
+
+    // Seems to eliminate some keepAlive timer that's making the bot auto-pause
+    const oldNetworking = Reflect.get(oldState, "networking");
+    const newNetworking = Reflect.get(newState, "networking");
+
+    const networkStateChangeHandler = (
+      oldNetworkState,
+      newNetworkState
+    ) => {
+      const newUdp = Reflect.get(newNetworkState, "udp");
+      clearInterval(newUdp?.keepAliveInterval);
+    };
+
+    oldNetworking?.off("stateChange", networkStateChangeHandler);
+    newNetworking?.on("stateChange", networkStateChangeHandler);
+  }).on("error", () => void 0);
     this.connection.subscribe(this.audioPlayer);
   }
   get channelId() {
